nimsuggest/nimsuggest.nim

```diff
@@ -599,14 +600,8 @@ proc handleCmdLine(cache: IdentCache; conf: ConfigRef) =
 
   if gMode != mstdin:
     conf.writelnHook = proc (msg: string) = discard
-  # Find Nim's prefix dir.
-  let binaryPath = findExe("nim")
-  if binaryPath == "":
-    raise newException(IOError,
-        "Cannot find Nim standard library: Nim compiler not in PATH")
-  conf.prefixDir = AbsoluteDir binaryPath.splitPath().head.parentDir()
-  if not dirExists(conf.prefixDir / RelativeDir"lib"):
-    conf.prefixDir = AbsoluteDir""
+
+  conf.prefixDir = AbsoluteDir(findNimRootCompileTime())
```

Comment:
Dubious, what if the Nim binary is in ``/usr/bin``? What has the compile-time location of Nim to do with anything?

Reply:
a common case is Nim is under /Users/timothee/.nimble//bin/nim ; in which case existing logic fails, but works under this PR; 
to make it robust, I guess we need a flag pointing to a Nim root repo (same value as reported by findNimRootCompileTime())
that flag can be (as other flags) set in ~/.config/nim/config.nims ; would that be acceptable?
there's other places where this could be useful

Reply:
1. we have the ``--lib`` flag to set the stdlib's location.
2. Nimsuggest is a variant of the Nim compiler. We use relative addressing to find the stdlib's location.
3. Nim(suggest) cannot be "built anywhere" with your PR either.

> that flag can be (as other flags) set in ~/.config/nim/config.nims ; would that be acceptable?

That's the --lib command line option, or maybe --path.